flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/RpcSystem.java

```diff
@@ -51,6 +51,9 @@ RpcServiceBuilder remoteServiceBuilder(
             @Nullable String externalAddress,
             String externalPortRange);
 
+    /** Hook to cleanup resources, like common thread pools or classloaders. */
+    void cleanup();
```

Comment:
Could we use closeable? Makes it easier for code analyzes to find leaks.

Reply:
sure

Reply:
One downside is that the call sites are harder to find via `find usages`.